pyccel/ast/core.py

```diff
@@ -398,16 +398,23 @@ class Allocate(PyccelAstNode):
         In C this provides the size which will be passed to malloc. In Fortran
         this provides the source argument of the allocate function.
 
+    alloc_type : str {'init'|'reserve'|'resize'}, optional
+        Defines the method of memory allocation exclusively for STC containers.
+        This parameter is irrelevant and should not be used for anything other than STC containers.
```

Comment:
I'm not a fan of this description. STC containers work very similarly to gFTL containers (and containers in other languages e.g. std containers in C++) so I'm not convinced that this parameter **should not be used for anything other than STC containers**

Reply:
fixed